### PR TITLE
New clarification prompt

### PR DESCRIPTION
**Pull request by lukaspetersson**
#708 and #703 

tldr: clarification flow is confusing for user and adds unnecessary information that confuses GPT.

Note: does not make the list of clarifications anymore.

I would love feedback on whether this could cause unexpected consequences.

### CODE REVIEW

File: gpt_engineer/steps.py [hunk 1]
```diff
@@ -76,12 +76,12 @@ def clarify(ai: AI, dbs: DBs) -> List[Message]:
...
         msg = messages[-1].content.strip()
 
-        if msg == "Nothing more to clarify.":
+        if "nothing to clarify" in msg.lower():
             break
 
         if msg.lower().startswith("no"):
-            print("Nothing more to clarify.")
             break
+            print("Nothing to clarify.")
```
Comment (AntonOsika) [inline gpt_engineer/steps.py:84]: Dead line?

Comment (lukaspetersson) [reply gpt_engineer/steps.py:83]: I apologize, fixed now

**Comment by AntonOsika** [comment]
I think clarify is very much a UX questions.

Are you confident this doesn't make it unstable/degrade behavior/etc?

We could also just go for #717 instead

**Comment by lukaspetersson** [comment]
> I think clarify is very much a UX questions.
> 
> Are you confident this doesn't make it unstable/degrade behavior/etc?
> 
> We could also just go for #717 instead

#717 just hides clarify, but it still leaves all the problems with it if user decides to use it. 

I ran it a couple of times (twice on 2 different cases) and I very much prefered the UX over the current version. It results in a shorter total prompt, which is beneficial I think, but this is just speculation and intuition.

**Comment by AntonOsika** [comment]
Cool let’s go

**Pull request merged**